File: ChatClientVB/Models/Partecipante.vb
Imports System.Collections.ObjectModel

Public Class Partecipante
    Inherits ViewModelBase

    Public Property Nome As String
    Public Property imgProfilo As Byte()
    Public Property Chatter As New ObservableCollection(Of MessaggioChat)

    Private _loggato As Boolean = True
    Public Property Loggato As Boolean
        Get
            Return _loggato
        End Get
        Set(value As Boolean)
            _loggato = value
            OnPropertyChanged()
        End Set
    End Property

    Private _haMandatoNuovoMsg As Boolean
    Public Property HaMandatoNuovoMsg As Boolean
        Get
            Return _haMandatoNuovoMsg
        End Get
        Set(value As Boolean)
            _haMandatoNuovoMsg = value
            OnPropertyChanged()
        End Set
    End Property

    Private _staScrivendo As Boolean
    Public Property StaScrivendo As Boolean
        Get
            Return _staScrivendo
        End Get
        Set(value As Boolean)
            _staScrivendo = value
            OnPropertyChanged()
        End Set
    End Property
End Class

File: ChatClientCS/Models/Partecipante.cs
using ChatClientCS.ViewModels;
using System.Collections.ObjectModel;

namespace ChatClientCS.Models
{
    public class Partecipante : ViewModelBase
    {
        public string Nome { get; set; }
        public byte[] imgProfilo { get; set; }
        public ObservableCollection<MessaggioChat> Chatter { get; set; }

        private bool _loggato = true;
        public bool Loggato
        {
            get { return _loggato; }
            set { _loggato = value; OnPropertyChanged(); }
        }

        private bool _haInviatoNuovoMessaggio;
        public bool HaInviatoNuovoMessaggio
        {
            get { return _haInviatoNuovoMessaggio; }
            set { _haInviatoNuovoMessaggio = value; OnPropertyChanged(); }
        }

        private bool _staScrivendo;
        public bool StaScrivendo
        {
            get { return _staScrivendo; }
            set { _staScrivendo = value; OnPropertyChanged(); }
        }

        public Partecipante() { Chatter = new ObservableCollection<MessaggioChat>(); }
    }
}